### PR TITLE
smartbond UART power management

### PR DESCRIPTION
**Pull request by kasjer**
This adds power management functionality to SmartBond(tm) serial driver.

With this PR:
- UART allows platform to go to sleep handling PM_DEVICE_ACTION_RESUME/PM_DEVICE_ACTION_SUSPEND messages.
- Platform can be woken up on UART RX activity
- Platform can be keep awaken when DTR line is asserted (optional feature of selected GPIO)
- DT now has definition of UART2 and UART3

This change requires #61857 and #61844 and #61941

Signed-off-by: Jerzy Kasenberg <jerzy.kasenberg@codecoup.pl>

### CODE REVIEW

**Comment by MaureenHelm** [comment]
@andrzej-kaczmarek @dcpleung please take a look

**Comment by kasjer** [comment]
Just reminder that this requires #61857

File: dts/arm/renesas/smartbond/da1469x.dtsi [hunk 1]
```diff
@@ -199,6 +199,24 @@
 			status = "disabled";
 		};
 
+		uart2: uart@50020100 {
```
Comment (dcpleung) [inline dts/arm/renesas/smartbond/da1469x.dtsi:202]: For this commit, could you amend the commit summary to include "smartbond" somewhere? For example, `dts: serial/smartbond: add uart2 and uart3 definitions`. Current summary will confuse people looking at the git log as it says adding uart2 and uart3 for all.

**Review by dcpleung — APPROVED** [review]

**Review by dcpleung — CHANGES_REQUESTED** [review]
(clicked the wrong button.)

File: dts/arm/renesas/smartbond/da1469x.dtsi (hunk 1 above)
Comment (kasjer) [reply dts/arm/renesas/smartbond/da1469x.dtsi:202]: fixed

**Review by dcpleung — DISMISSED** [review]

**Comment by kasjer** [comment]
Still waiting for #61857 to be processed

**Comment by kasjer** [comment]
Still waiting for #61857 to be accepted

**Comment by nashif** [comment]
needs rebase

File: drivers/serial/uart_smartbond.c [hunk 2]
```diff
@@ -238,11 +311,104 @@ static int uart_smartbond_config_get(const struct device *dev,
...
+	}
+	if (rx_wake_timeout > 0 && config->rx_wake_gpio.port != NULL) {
+		k_work_init_delayable(&data->rx_timeout_work,
+				      uart_smartbond_rx_refresh_timeout);
+		gpio_init_callback(&data->rx_wake_cb, uart_smartbond_wake_handler,
+				   BIT(config->rx_wake_gpio.pin));
+
+		ret = gpio_add_callback(config->rx_wake_gpio.port, &data->rx_wake_cb);
+		if (ret < 0) {
+			return ret;
```
Comment (ydamigos) [inline drivers/serial/uart_smartbond.c:400]: Should we call `da1469x_pd_release(MCU_PD_DOMAIN_COM);` before returning?

**Review by ydamigos — CHANGES_REQUESTED** [review]

File: drivers/serial/uart_smartbond.c (hunk 2 above)
Comment (kasjer) [reply drivers/serial/uart_smartbond.c:400]: @ydamigos you are right, code will be changed not to return here. If add callback fails (unlikely) timeout will not be configured and UART will work except timeout functionality will be inactive

**Comment by kasjer** [comment]
rebased to fix conflicts and @ydamigos comment addressed

**Review by ydamigos — APPROVED** [review]

**Review by andrzej-kaczmarek — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**